src/adapter/etl-adapter-google-sheet/tests/Flow/ETL/Adapter/GoogleSheet/Tests/Integration/GoogleSheetTest.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Flow\ETL\Adapter\GoogleSheet\Tests\Integration;
+
+use function Flow\ETL\Adapter\GoogleSheet\{from_google_sheet,
+    google_create_spreadsheet,
+    google_sheets,
+    to_google_sheet};
+use function Flow\ETL\DSL\{config, flow_context, int_entry, row, rows, str_entry, string_entry};
+use Flow\ETL\Tests\FlowTestCase;
+use Google\Service\Sheets;
+use Google\Service\Sheets\ClearValuesRequest;
+
+final class GoogleSheetTest extends FlowTestCase
+{
+    private const SHEET_NAME = 'Flow Sheet';
+
+    private Sheets $service;
+
+    private string $spreadsheetId;
+
+    protected function setUp() : void
+    {
+        parent::setUp();
+
+        if (!isset($_ENV['GOOGLE_SHEETS_AUTH'])) {
+            self::markTestSkipped('GOOGLE_SHEETS_AUTH environment variable is not set');
+        }
+
+        try {
+            $this->service = google_sheets(
+                json_decode((string) $_ENV['GOOGLE_SHEETS_AUTH'], true, 512, JSON_THROW_ON_ERROR),
+                Sheets::SPREADSHEETS
+            );
+        } catch (\JsonException) {
+            self::markTestSkipped('GOOGLE_SHEETS_AUTH environment variable is not valid JSON');
+        }
+
+        $this->spreadsheetId = google_create_spreadsheet(
+            $this->service,
+            'Flow test spreadsheet',
+            self::SHEET_NAME,
+        );
+
+        $this->clearTestSheet();
+    }
+
+    protected function tearDown() : void
+    {
+        parent::tearDown();
+
+        $this->clearTestSheet();
+    }
+
+    public function test_load_and_extract() : void
```

Comment:
```bash
/usr/local/Cellar/php@8.2/8.2.28_1/bin/php/Users/stloyd/Documents/flow/tools/phpunit/vendor/phpunit/phpunit/phpunit --configuration /Users/stloyd/Documents/flow/phpunit.xml.dist --filter Flow\\ETL\\Adapter\\GoogleSheet\\Tests\\Integration\\GoogleSheetTest --test-suffix GoogleSheetTest.php /Users/stloyd/Documents/flow/src/adapter/etl-adapter-google-sheet/tests/Flow/ETL/Adapter/GoogleSheet/Tests/Integration --teamcity
Testing started at 11:23 ...
PHPUnit 11.5.18 by Sebastian Bergmann and contributors.

Runtime:       PHP 8.2.28
Configuration: /Users/stloyd/Documents/flow/phpunit.xml.dist
Time: 00:05.952, Memory: 16.00 MB

OK (1 test, 7 assertions)
Process finished with exit code 0

Reply:
it's with auth env var set?

Reply:
Replaced with a file-based one as we spoke before.